packages/cli/src/fulltext/querySymbols.ts

```diff
@@ -0,0 +1,74 @@
+import { readFile } from 'fs/promises';
+import queryKeywords from './queryKeywords';
+
+export const SymbolRegexes: Record<string, RegExp[]> = {
+  cs: [
+    /(interface|class|enum|struct)\s+(?<symbol>[\w\d_]+)/gm,
+    /(?<symbol>[\w\d_~]+)\s*?\(.*?\)\s*?\{/g,
+  ],
+  cpp: [
+    /(struct|enum|union|class)\s+(?<symbol>[\w\d_]+)\s*?\{/g,
+    /}\s*?(?<symbol>[\w\d_]+)\s*?;/g,
+    /(?<symbol>[\w\d_~]+)\s*?\(.*?\)\s*?\{/gm,
+  ],
+  rs: [
+    /(struct|enum|union|trait|type)\s+(?<symbol>[\w\d_]+)/gm,
+    /fn\s+?(?<symbol>[\w\d_~]+)\s*?\(/gm,
+  ],
+  go: [/type\s+(?<symbol>[\w\d_]+)/gm, /func\s+?(\(.*?\)\s*?)?(?<symbol>[\w\d_~]+)\s*?\(/gm],
+  rb: [/(class|module)\s+(?<symbol>[\w\d_]+)/gm, /def\s+?(?<symbol>[\w\d_~]+)/gm],
+  py: [/(class|def)\s+(?<symbol>[\w\d_]+)/gm],
+  java: [
+    /(class|@?interface|enum)\s+(?<symbol>[\w\d_]+)/gm,
+    /(?<symbol>[\w\d_~]+)\s*?\(.*?\)\s*?\{/gm,
+  ],
+  ts: [
+    /(class|interface|enum|type)\s+(?<symbol>[\w\d_]+)/gm,
+    /(?<symbol>[\w\d_~$#]+)\s*?\(.*?\)\s*?\{/gm,
+    /(?<symbol>[\w\d_~$#]+)\s*?=\s*?\(.*?\)\s*?=>/gm,
+  ],
+  kt: [
+    /(class|typealias)\s+(?<symbol>[\w\d_]+)/gm,
+    /fun\s+?(<.+?>\s+)?(.*?\.)?(?<symbol>[\w\d_~]+)/gm,
+  ],
+  php: [/(class|trait|function)\s+(?<symbol>[\w\d_]+)/gm],
+};
+
+// Define aliases for common file extensions
+['js', 'jsx', 'ts', 'tsx', 'vue', 'svelte'].forEach((ext) => {
+  SymbolRegexes[ext] = SymbolRegexes.ts;
+});
```

Comment:
We probably want a generic one too.

Reply:
I was thinking that too, but I'm not sure what a generic syntax would be. In many cases there are similarities and subtle differences. Ultimately I decided it might be better to fall back on the current behavior than attempt to parse with unexpected results.

For language coverage, I'm using this as reference:
https://survey.stackoverflow.co/2023/#most-popular-technologies-language

Another option would be to expand support to more languages like Lua, Dart, Swift, etc., but that's getting into single digit usage. I skipped over SQL, bash, PowerShell, but perhaps it's worth considering them, e.g., table names as terms?

```regex
/(TABLE|FROM|INTO) "?(.*?)"?[\s;]/gi
```

Anyways, I'm open to ideas. A generic parser would be worthwhile but I haven't struck a clear solution.

Reply:
Just to close the loop on this, I did end up adding generic language support. In testing it's validating it can pick up support for Dart, Swift, GDScript and others.